### ISSUE TITLE
[[clang::lifetimebound]] doesn't work with default arguments

### ISSUE THREAD

**Issue by higher-performance** [llvm/llvm-project#68596]
```cpp
const int& foo(const int& x [[clang::lifetimebound]] = 0) {
    return x;
}
int main() {
    const int& x = foo();  // No error
    return x;
}
```

**Comment by shafik** [comment]
I thought maybe this was working as intended since the lifetime of the default param would be in the function that called `foo` but an extended example also shows this behavior: https://godbolt.org/z/G1b9b1Ez6

CC @zygoloid who implemented the feature via f4e248c23e05e

**Comment by zygoloid** [comment]
This seems like an oversight to me. Might be as simple as stepping through `CXXDefaultArgExpr`s at the right place.
